src/v/cluster/partition_balancer_planner.cc

```diff
@@ -890,10 +910,10 @@ void partition_balancer_planner::get_rack_constraint_repair_actions(
 
     for (const auto& ntp : ctx.state().ntps_with_broken_rack_constraint()) {
```

Comment:
so `co_await ctx.with_partition` yields, and this means we need to guard against changes in `ntps_with_broken_rack_constraint` (which can change under our feet with concurrent topic updates)